File: src/ch05/ApiMySQL/Models/blazordbContext.cs
using System;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata;
using MySql.EntityFrameworkCore;


#nullable disable

namespace ApiMySQL.Models
{
    public partial class blazordbContext : DbContext
    {
        public blazordbContext()
        {
        }

        public blazordbContext(DbContextOptions<blazordbContext> options)
            : base(options)
        {
        }

        public virtual DbSet<WpPost> WpPosts { get; set; }
        public virtual DbSet<WpUser> WpUsers { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            if (!optionsBuilder.IsConfigured)
            {
                // optionsBuilder.UseMySQL("server=localhost;user id=wpuser;persistsecurityinfo=True;database=wpdb");
            }
        }
    }
}

File: src/ch04/wpfvb/blazordbContext.vb
Imports Microsoft.EntityFrameworkCore

Public Class blazordbContext
    Inherits DbContext

    Public Sub New()

    End Sub

    Public Sub New(options As DbContextOptions(Of blazordbContext))
        MyBase.New(options)
    End Sub

    Public Overridable Property Books As DbSet(Of Book)

    Protected Overrides Sub OnConfiguring(optionsBuilder As DbContextOptionsBuilder)
        If Not optionsBuilder.IsConfigured Then
            optionsBuilder.UseSqlServer("Server=.;Database=blazordb;Trusted_connection=True")
        End If
    End Sub

End Class
